augur/utils.py

```diff
@@ -84,16 +84,26 @@ def is_date_ambiguous(date, ambiguous_by="all"):
     ambiguous_by : str
         Field of the date string to test for ambiguity ("day", "month", "year", "all")
     """
-    year, month, day = date.split('-')
+    date_components = date.split('-', 2)
+
+    if len(date_components) == 3:
+        year, month, day = date_components
+    elif len(date_components) == 2:
+        year, month = date_components
+        day = "XX"
+    else:
+        year = date_components[0]
+        month = "XX"
+        day = "XX"
+
     return (
-            (ambiguous_by == 'all' and 'X' in date) or
+            (ambiguous_by == 'all' and ('X' in year or 'X' in month or 'X' in day)) or
             (ambiguous_by == 'day' and 'X' in day) or
```

Comment:
To go with the tests I added (below), I suggest the logic here is changed to
```
    return (
        (ambiguous_by == 'all' and ('X' in year or 'X' in month or 'X' in day)) or
        (ambiguous_by == 'day' and ('X' in year or 'X' in month or 'X' in day)) or
        (ambiguous_by == 'month' and ('X' in year or 'X' in month)) or
        (ambiguous_by == 'year' and ('X' in year))
    )
```

Reply:
Excellent. This helped me realize the redundancy in this original approach to the tests. I've included your logic here in the original code and then refactored it to simplify the logic.